### ISSUE TITLE
Problem parsing the -X option of dirac-pilot.py from the CS

### ISSUE THREAD

**Issue by arrabito** [DIRACGrid/DIRAC#3462]
I just discovered that, one cannot use the ExtraPilotOptions in the CS to pass a customized list of pilot commands using the -X option.
Indeed the list of commands is interpreted like a CS option of a list type, which results in passing only the first argument of the list to the dirac-pilot.py script.
I was wondering which would be the correct way to handle this use-case.

**Comment by fstagni** [comment]
This use case is only treated "correctly" in Pilot3: with Pilot2 there is indeed no possibility, IIRC, to pass the list of commands in the CS, while with Pilot3 the list of commands can even be different depending on the type of CE.

In LHCb we use anyway LHCb-specific commands (they can be found here: https://gitlab.cern.ch/lhcb-dirac/LHCbDIRAC/blob/master/LHCbDIRAC/WorkloadManagementSystem/PilotAgent/LHCbPilotCommands.py). What we have done is that we have created a very simple extension of the SiteDirector, where we replaced have added the LHCb specific commands. It can be found in https://gitlab.cern.ch/lhcb-dirac/LHCbDIRAC/blob/master/LHCbDIRAC/WorkloadManagementSystem/Agent/SiteDirector.py#L123
